platform/mbed_enum_flags.h

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2020 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_SCOPED_ENUM_FLAGS_H
+#define MBED_SCOPED_ENUM_FLAGS_H
+
+#include <type_traits>
+
+#define ENUM_FLAG_OPERATOR(T, OP) \
+inline constexpr T operator OP( T lhs, T rhs) \
```

Comment:
Seems astyle is being lenient with you, presumably because of the macros. Quite a few excess spaces inside parentheses and `<>`. Here and in some other places.

Reply:
I'm just making sure the macro expansion actually expands macros correctly. I will remove every space to make it virtually the same as a normal function.

Reply:
I can't think of any way spaces and macros interact. Tokenisation happens before macro substitution of tokens. There's no way tokens can get "accidentally" pasted together.